### PR TITLE
Revert to single arch-specific package (Fix #22)

### LINKED ISSUE #22: Installation on Mac with M1
Could you make a new build available of version 0.7.0 such that it is possible to install it under Mac?

Conda gives me the following messages:
```
conda search  -c ome bioformats2raw
Loading channels: done
# Name                       Version           Build  Channel             
bioformats2raw                 0.2.1               1  ome                 
bioformats2raw                 0.2.2               2  ome 
```
and
```
conda install -y -c ome 'bioformats2raw==0.7.0'
Channels:
 - ome
 - defaults
 - conda-forge
Platform: osx-arm64
Collecting package metadata (repodata.json): done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - bioformats2raw==0.7.0

Current channels:

  - https://conda.anaconda.org/ome
  - defaults

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

Downloading and installing [the latest zip file](https://github.com/glencoesoftware/bioformats2raw/releases/tag/v0.7.0) does not work either, because then I get a `Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.UnsatisfiedLinkError: Unable to load library 'blosc'` error that I cannot resolve.

Thank you.

### PR DESCRIPTION
**Pull request by joshmoore**
~Previously, a special patch was needed on Windows to handle the long jar list. With this patch removed, the main build can again be architecture-neutral.~

Updated `conda build` no longer handles the split packages (reason unknown). This temporarily merges them back into a single build. If anyone has an idea of how to separate them back out, happen to see it!

The original motivation for changing the conda setup task was the lack of support in s-wiegand for the new Mac M! runner.

cc: @jni @jburel @dgault

### CODE REVIEW

**Comment by manics** [comment]
If you wanted to test this fully (bearing in mind https://github.com/ome/conda-bioformats2raw/issues/22#issuecomment-1845007375) you'd need to build this under a single architecture, upload it as an artifact, then download and test the built package on multiple architectures.

**Comment by jni** [comment]
Kind ping... 🙏

**Comment by joshmoore** [comment]
The ping is appreciated, but unfortunately, I think this still isn't going to work.

*Either* this PR needs a way to launch on Windows & Posix without separate .bat & .sh files, *or* we need an OSX/arm runner (and nothing really has changed on that front).

Unless I'm missing something, anyone?

**Comment by manics** [comment]
That sounds right. The `bld.bat` and `build.sh` in https://github.com/ome/conda-bioformats2raw/tree/master/bioformats2raw are automatically called depending on the build platform (`bld.bat` when built on Windows, `build.sh` when built on Linux or OS X)- this is why the built packages are different. If it's only built on one platform (as implied by noarch) the resulting package will need to contain both Windows and Linux startup scripts.

If you wanted to be really thorough with testing (not saying you need to be!) you could build it on Linux, [upload the artifact](https://github.com/actions/upload-artifact), then have a follow-on matrix job that runs on all three platforms, [downloads the artifact](https://github.com/actions/download-artifact), and tests it.

**Comment by jni** [comment]
> or we need an OSX/arm runner

@joshmoore is this relevant???

https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/

**Comment by joshmoore** [comment]
I think so! This is what we've been waiting for.

**Comment by joshmoore** [comment]
Hmmm..... well, it's green. Thoughts?

One from my side is that we should really just be building the noarch part once, but I'll leave that for another day.

File: bioformats2raw/meta.yaml [hunk 1]
```diff
@@ -6,18 +6,16 @@ package:
...
   url: https://github.com/glencoesoftware/{{ name }}/releases/download/v{{ version
     }}/{{ name }}-{{ version }}.zip
   sha256: e94e3fd268bd144069d672899ee0b53182d2b3d71519d2fade8bfaedd6c7bf24
 
 requirements:
-  build:
-  - m2-patch    # [win]
   run:
-  - bioformats2raw-libs={{ PKG_VERSION }}-{{ PKG_BUILDNUM }}
+  - bioformats2raw-libs
```
Comment (manics) [inline bioformats2raw/meta.yaml]: The version pin is needed to ensure `conda upgrade bioformats2raw` automatically upgrades the `-libs` package in sync.

Comment (joshmoore) [reply bioformats2raw/meta.yaml]: Fair enough. Building them together, locally, though is no longer working. Any ideas on how to fix that? None of the following worked:

* `-c local`
* `--use-local`
* `conda build biofornats2raw{-libs,}` (i.e. at the same time)
* `... --output-folder X` (i.e. into the same folder)

Comment (manics) [reply bioformats2raw/meta.yaml]: Dunno, maybe you need this?
```
    defaults:
      run:
        shell: bash -el {0}
```
I've no idea why but it's in the README:
https://github.com/conda-incubator/setup-miniconda?tab=readme-ov-file#usage-examples

Comment (joshmoore) [reply bioformats2raw/meta.yaml]: It's failing even locally just with `conda build`.

Comment (joshmoore) [reply bioformats2raw/meta.yaml]: Would the use of `outputs:` be a more modern version of what's going on here?

e.g., https://github.com/AnacondaRecipes/boost-feedstock/blob/master/recipe/meta.yaml

**Comment by joshmoore** [comment]
I've failed all kinds of ways to get the two package version to build. I've remerged them.

**Comment by joshmoore** [comment]
Build is now green. I'm going to merge and tag with `rc` so I can test with a `--pre` flag, then we should be ready for the final sign off on this.

**Comment by jni** [comment]
Cool! Lemme know if you need some testers! 😊

**Pull request merged**